core/server/worker/src/main/java/alluxio/worker/block/reviewer/AllocationCoordinator.java

```diff
@@ -0,0 +1,52 @@
+package alluxio.worker.block.reviewer;
+
+import alluxio.worker.block.BlockMetadataView;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.allocator.Allocator;
+import alluxio.worker.block.meta.StorageDirView;
+
+public class AllocationCoordinator {
+  public Reviewer mReviewer;
+  public Allocator mAllocator;
+
+  private static AllocationCoordinator sInstance = null;
+
+  private AllocationCoordinator(BlockMetadataView view) {
+    mAllocator = Allocator.Factory.create(view);
+    mReviewer = Reviewer.Factory.create();
+  }
+
+  public static AllocationCoordinator getInstance(BlockMetadataView view) {
+    if (sInstance == null) {
+      synchronized (AllocationCoordinator.class) {
+        if (sInstance == null) {
+          sInstance = new AllocationCoordinator(view);
+        }
+      }
+    }
+    return sInstance;
+  }
+
+  /**
+   * This is currently only used by tests, where the singleton state needs to be reconfigured.
+   * */
+  public static synchronized void destroyInstance() {
+    sInstance = null;
+  }
+
+  public StorageDirView allocateBlockWithView(long sessionId, long blockSize, BlockStoreLocation location,
+                                       BlockMetadataView view, boolean skipReview) {
+    if (skipReview) {
+      return mAllocator.allocateBlockWithView(sessionId, blockSize, location, view, v -> true);
+    }
+    return mAllocator.allocateBlockWithView(sessionId, blockSize, location, view, mReviewer::acceptAllocation);
```

Comment:
Although we can do this, I still DO NOT find a way to completely remove `AcceptingReviewer`. The reason is some integration tests:
```
// This is the levels
LocalAlluxioCluster
  WorkerProcess
    TieredBlockStore
      TieredBlockStore.allocateSpace
        TieredBlockStore.mAllocationCoordinator.allocateBlockWithView with true or false, depending on the logic in allocateSpace
```

In the integration tests we are creating real worker processes, there's no way to force the allocator to skip all reviews, except having an `AcceptingReviewer`. With that said, `AcceptingReviewer` class is ONLY NEEDED by the integration tests.

I also tried to move `AcceptingReviewer` into the `alluxio-test` module, but that DOES NOT work, because the worker module does NOT put the `alluxio-test` module on the classpath so can NOT find the `AcceptingReviewer` if that is in the test module. 

Continue on that thread, we are not really enabling people to extend our Allocator interface. Imagine I'm a user, the ONLY way to extend Allocator is to put my implementation in the `alluxio-worker` module (or the ones it depends on, essentially alluxio core modules), and recompile Alluxio. It's not really possible to plug my class in as a jar or something, despite the fact that we have `PropertyKey,ALLOCATOR_CLASS` which looks very extendable. @gpang I think it makes sense to change to service loader, does that make sense to you?

Reply:
I see. I don't expect the reviewer to be extended anytime soon, so we can just keep it like this for now.